docs/getting-started/setup.md

```diff
@@ -1,4 +1,144 @@
-## Installation
+## Prerequisites
+
+### Python
+- **Supported versions:** 3.9, 3.10, 3.11, 3.12 (recommended: latest 3.12.x)
+- **Check version:**
+```sh
+    python3 --version
+```
+- **Install Python:** You can install Python by downloading it from [python.org](https://www.python.org/downloads/) or by using a version manager such as [pyenv](https://github.com/pyenv/pyenv).
+
+### Java
+- **Supported versions:** 8, 11, 17 (recommended: latest 17.x)
+- **Check version:**
+```sh
+  java -version
+```
+- **Install Java:** You can install Java by downloading it from [Oracle](https://www.oracle.com/java/technologies/javase-jdk11-downloads.html) or by using a package manager such as [SDKMAN](https://sdkman.io/) or [OpenJDK](https://openjdk.org/).
+- **For macOS users:** You can install OpenJDK using Homebrew:
+```sh
+  brew install openjdk@17
+```
+Add to your shell config:
+```sh
+  export JAVA_HOME="/opt/homebrew/opt/openjdk@17"
+  export PATH="$JAVA_HOME/bin:$PATH"
+```
```

Comment:
Instead of providing detailed steps for installing Java I would leave it at required version and recommendation of using JDK version manager like SDKMan or equivalent. Same for inclusion of setting JAVA_HOME on the PATH